server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationStatsIT.java

```diff
@@ -19,34 +19,51 @@
 
 import java.util.concurrent.CountDownLatch;
 import static java.util.Arrays.asList;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
 
 @OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0)
 public class SegmentReplicationStatsIT extends SegmentReplicationBaseIT {
 
     public void testSegmentReplicationStatsResponse() throws Exception {
-        final String primaryNode = internalCluster().startNode();
-        createIndex(INDEX_NAME);
-        ensureYellowAndNoInitializingShards(INDEX_NAME);
-        final String replicaNode = internalCluster().startNode();
-        ensureGreen(INDEX_NAME);
-
-        // index 10 docs
-        for (int i = 0; i < 10; i++) {
-            client().prepareIndex(INDEX_NAME).setId(Integer.toString(i)).setSource("field", "value" + i).execute().actionGet();
+        internalCluster().startClusterManagerOnlyNode();
+        String dataNode = internalCluster().startDataOnlyNode();
+        String anotherDataNode = internalCluster().startDataOnlyNode();
+
+        int numShards = 4;
+        assertAcked(
+            prepareCreate(
+                INDEX_NAME,
+                0,
+                Settings.builder()
+                    .put("number_of_shards", numShards)
+                    .put("number_of_replicas", 1)
+                    .put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT)
+            )
+        );
+        ensureGreen();
+        final long numDocs = scaledRandomIntBetween(50, 100);
+        for (int i = 0; i < numDocs; i++) {
+            index(INDEX_NAME, "doc", Integer.toString(i));
         }
         refresh(INDEX_NAME);
-        waitForSearchableDocs(10L, asList(primaryNode, replicaNode));
+        ensureSearchable(INDEX_NAME);
 
         assertBusy(() -> {
```

Comment:
As I mentioned [here](https://github.com/opensearch-project/OpenSearch/pull/6268#discussion_r1112410862), seen one instance where this test failed. May be we can increase the assertBusy timeout and see if it helps.